### ISSUE TITLE
Autocomplete for attachments/notes/tags

### ISSUE THREAD

**Issue by vorg** [fabiospampinato/notable#20]
It would be super useful to have autocomplete for attachments / notes / tags like here https://codemirror.net/demo/complete.html

<img width="592" alt="screenshot 2018-12-27 at 18 23 09" src="https://user-images.githubusercontent.com/171001/50474500-7aecd880-0a04-11e9-9f8f-903ada6dbbe2.png">

Currently I need to click attachments button, memorise the name of the file and then type it manually.

Alternative solution is to implement dragging images/notes/tags into text editor from Finder/Middlebar/Sidebar

**Comment by fabiospampinato** [comment]
Definitely, I should probably add an autocomplete widget to [Svelto](https://github.com/svelto/svelto) (the UI framework Notable uses) first.

**Comment by fabiospampinato** [comment]
Tags autocompletion should also be available when editing them, and in the future also when searching #106

**Comment by mikaelmello** [comment]
Came here to ask for this feature and found an existing issue. Great! +1. I'll be watching this issue and sorry that I can't contribute

**Comment by mjedmonds** [comment]
Seconding this!

**Comment by luisduenas** [comment]
Same here, is there any way i can help with this feature?

**Comment by fabiospampinato** [comment]
@luisduenas in order to implement this I first need to add an autocomplete widget to [Svelto](https://github.com/svelto/svelto), which unfortunately it's still undocumented, so I'll probably have to do that myself.

**Comment by amelandri** [comment]
I second this. 

After importing all my notes from Evernote, I had to use vscode to insert images references in markdown file, because it was too hard to memorize the image name and then write it down.

**Comment by wAuner** [comment]
tag selection (like dropdown or something) and autocomplete for tags would be great!
having to type out a nested tag is quite error prone.

**Comment by mishaal79** [comment]
Please add in a feature to copy and paste screenshots from the clipboard.

**Comment by fabiospampinato** [comment]
> Please add in a feature to copy and paste screenshots from the clipboard.

@mishaal79 That's a very requested feature that's tracked [here](https://github.com/notable/notable/issues/26).

**Comment by ChuckJonas** [comment]
yes please (I was just going to suggest to make them case-insensitive, but this would work as well)

**Comment by fabiospampinato** [comment]
We can now copy-paste attachments/notes/tags, thanks to @Armagedoom's PR! https://www.reddit.com/r/Notable/comments/bmm5ub/preview_copypaste_support_for_attachmentsnotestags/

Autocomplete will be more convenient than this, but at least the situation improved a bit.

**Comment by rathboma** [comment]
++ to this. I've copied over a bunch of MD files and the filenames are different to the titles, so copy/paste doesn't work as expected. Autocomplete would totally solve that

**Comment by H8to** [comment]
You could also try implementing drag and drop notes to tags or tags to notes. That would ease the process a lot!
Maybe that's also helpful idea as interim solution to auto-completion?

**Comment by fabiospampinato** [comment]
@H8to Yes, we are tracking there here: #15

**Comment by ejcjason** [comment]
How is this going on now? It would be really helpful to have autocomplete function for tags. Do you need any help for that?

**Comment by fabiospampinato** [comment]
@ejcjason This hasn't been implemented yet, I'm working on something else right now. Thanks but I don't think I need help to implement this, if you want to help with some coding work though email me at fabio@notable.md as there probably are little standalone projects I could use some more helping hands with.

**Comment by sanan-fataliyev** [comment]
autocomplete for tags please! (and thank you for this awesome app)

**Comment by mikejones3** [comment]
Def need this feature.

**Comment by broox** [comment]
+1 - this issue might be easier to tackle if broken out into different, smaller chunks? for my use-case, i'd just like the ability to autocomplete tags when tagging a given note.

P.S. I'm loving the simple UI and the file-based approach of this app so far.

**Comment by bjankord** [comment]
+1 for this feature, its the main feature that is missing for me.

**Comment by kanlukasz** [comment]
It's been 6 years already, is there any real chance of implementing this?
